### PR TITLE
PP-6614 Normalise the last four digits for Apple Pay

### PR DESCRIPTION
**Pull request by narinderchana**
Description:
- If the displayName is less than four characters then map the last four digits to an empty string.
- If the last four characters of the displayName are not numeric then map the last four digits to an empty string.

### CODE REVIEW

File: test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js [hunk 1]
```diff
@@ -53,6 +53,120 @@ describe('normalise apple pay payload', function () {
...
+          data: 'MLHhOn2BXhNw9wLLDR48DyeUcuSmRJ6KnAIGTMGqsgiMpc+AoJ…LUQ6UovkfSnW0sFH6NGZ0jhoap6LYnThYb9WT6yKfEm/rDhM=',
+          signature: 'MIAGCSqGSIb3DQEHAqCAMIACAQExDzANBglghkgBZQMEAgEFAD…ZuQFfsLJ+Nb3+7bpjfBsZAhA1sIT1XmHoGFdoCUT3AAAAAAAA',
+          header: {
+            ephemeralPublicKey: 'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE5/Qc6z4TY5HQ5n…KC3kJ4DtIWedPQ70N35PBZzJUrFjtvDZFUvs80uo2ynu+lw==',
+            publicKeyHash: 'Xzn7W3vsrlKlb0QvUAviASubdtW4BotWrDo5mGG+UWY=',
+            transactionId: '372c3858122b6bc39c6095eca2f994a8aa012f3b025d0d72ecfd449c2a5877f9'
+          }
+        },
+        paymentMethod: {
+          displayName: 'JBC',
```
Comment (alexbishop1) [inline test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js:115]: Can we make this something like `123`? That way we know we’re definitely hitting the length restriction in this test and not the has-to-be-all-digits restriction.

File: test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js [hunk 2]
```diff
@@ -53,6 +53,120 @@ describe('normalise apple pay payload', function () {
...
+          data: 'MLHhOn2BXhNw9wLLDR48DyeUcuSmRJ6KnAIGTMGqsgiMpc+AoJ…LUQ6UovkfSnW0sFH6NGZ0jhoap6LYnThYb9WT6yKfEm/rDhM=',
+          signature: 'MIAGCSqGSIb3DQEHAqCAMIACAQExDzANBglghkgBZQMEAgEFAD…ZuQFfsLJ+Nb3+7bpjfBsZAhA1sIT1XmHoGFdoCUT3AAAAAAAA',
+          header: {
+            ephemeralPublicKey: 'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE5/Qc6z4TY5HQ5n…KC3kJ4DtIWedPQ70N35PBZzJUrFjtvDZFUvs80uo2ynu+lw==',
+            publicKeyHash: 'Xzn7W3vsrlKlb0QvUAviASubdtW4BotWrDo5mGG+UWY=',
+            transactionId: '372c3858122b6bc39c6095eca2f994a8aa012f3b025d0d72ecfd449c2a5877f9'
+          }
+        },
+        paymentMethod: {
+          displayName: '       ',
```
Comment (alexbishop1) [inline test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js:153]: This is actually blank, not empty.

File: test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js (hunk 1 above)
Comment (narinderchana) [reply test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js]: Change has been made

File: test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js (hunk 2 above)
Comment (narinderchana) [reply test/controllers/web-payments/apple-pay/normalise-apple-pay-payload.test.js:153]: Change has been made

**Pull request merged**